Classes/Domain/Service/JsonService.php

```diff
@@ -0,0 +1,235 @@
+                continue;
+            }
+            foreach ($cfg['elements'] as $elKey => $elCfg) {
+                if (empty($elCfg['columns'])) {
+                    continue;
+                }
+
+                // initialize create module
+                $module = array();
+                $module[$table]['elements'][$elKey] = $elCfg;
+                $module[$table]['sql'] = array_intersect_key($cfg['sql'], array_flip($elCfg['columns']));
+                $module[$table]['tca'] = array_intersect_key($cfg['tca'], array_flip($elCfg['columns']));
+
+                // add additional tables based on column names
+                foreach ($elCfg['columns'] as $column) {
+                    if (isset($data[$column])) {
+                        $module[$column] = $data[$column];
+                    }
+                }
+
+                // get all database fields from current module tables
+                $currentFields = array();
+                foreach ($module as $modCfg) {
+                    if (isset($modCfg['sql'])) {
+                        $currentFields = array_merge($currentFields, array_keys($modCfg['sql']));
+                    }
+                }
+
+                // find used sys_file_references and add them to module data
+                $references = array_filter($data['sys_file_reference']['sql'], function ($key) use ($currentFields) {
+                    return in_array($key, $currentFields);
+                }, ARRAY_FILTER_USE_KEY);
+                if (!empty($references)) {
+                    $module['sys_file_reference']['sql'] = $references;
+                }
+
+                $modules[$table . '_' . $elKey] = $module;
+            }
+        }
+
+        return $modules;
+    }
+
+    /**
+     * Recursively computes the intersection of arrays using keys for comparison.
+     *
+     * @param   array $array1 The array with master keys to check.
+     * @param   array $array2 An array to compare keys against.
+     *
+     * @return  array associative array containing all the entries of array1 which have keys that are present in array2.
+     **/
+    private function searchForMergeError(array $array1, array $array2)
+    {
+        $flatArray1 = ArrayUtility::flatten($array1);
+        $flatArray2 = ArrayUtility::flatten($array2);
+        $intersection = array_intersect_key($flatArray1, $flatArray2);
+
+        $errors = array();
+        foreach ($intersection as $key => $value) {
+            if ($value == $flatArray2[$key]) {
+                unset($intersection[$key]);
+            } else {
+                $errors[$key] = array(
+                    'current'    => $value,
+                    'mergeValue' => $flatArray2[$key],
+                );
+            }
+        }
+
+        return $errors;
+    }
+
+
+    /**
+     * Return JSON formatted in PHP 5.4.0 and higher
+     *
+     * @param $data
+     *
+     * @return string
+     */
+    private function encodeJSON($data)
```

Comment:
Current master has no constraint regarding the PHP version. There is [no explicit PHP version required](https://github.com/Gernott/mask/blob/master/composer.json). Implicitly the following PHP versions apply:
* if this extension is used together with [TYPO3 v7.6 then the PHP version must be `>=5.5`](https://github.com/TYPO3/TYPO3.CMS/blob/TYPO3_7-6/composer.json#L31)
* if this extension is used together with [TYPO3 v8.7 then the PHP version must be `^7.0`](https://github.com/TYPO3/TYPO3.CMS/blob/TYPO3_8-7/composer.json#L34)

That means that this extension must support the following PHP versions: `>=5.5||^7.0` = `>=5.5`.

I would propse to define the required PHP version in the `composer.json` of this project. But this is beyond the scpoe of this Pull Request.